### PR TITLE
Used format strings throughout TARDIS

### PR DESCRIPTION
**Pull request by WinstonPais**
## Motivation and Context

#1205 
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have assigned/requested two reviewers for this pull request.

### CODE REVIEW

File: tardis/io/atom_data/base.py [hunk 1]
```diff
@@ -481,18 +475,11 @@ def _check_selected_atomic_numbers(self):
...
             raise AtomDataMissingError(msg)
 
     def __repr__(self):
-        return "<Atomic Data UUID={} MD5={} Lines={:d} Levels={:d}>".format(
-            self.uuid1,
-            self.md5,
-            self.lines.line_id.count(),
-            self.levels.energy.count(),
-        )
+        return f"<Atomic Data UUID={self.uuid1} MD5={self.md5} Lines={self.lines.line_id.count():d} Levels={self.levels.energy.count():d}>"
```
Comment (wkerzendorf) [inline tardis/io/atom_data/base.py:482]: PEP8 violation - use black to fix.

File: tardis/io/atom_data/base.py [hunk 2]
```diff
@@ -277,10 +273,8 @@ def _check_related(self):
 
             if len(check_list) != 0 and len(check_list) != len(group):
                 raise AtomDataMissingError(
-                    "The following dataframes from the related group [{0}] "
-                    "were not given: {1}".format(
-                        ", ".join(group), ", ".join(check_list)
-                    )
+                    f"The following dataframes from the related group [{', '.join(group)}] "
+                    f"were not given: {', '.join(check_list)}"
```
Comment (wkerzendorf) [inline tardis/io/atom_data/base.py:277]:
```suggestion
                    group_str = ', '.join(group)
                    check_list_str = ', '.join(check_list)
                    f"The following dataframes from the related group [{group_str}] "
                    f"were not given: {check_list_str}"
```

File: tardis/io/atom_data/base.py [hunk 3]
```diff
@@ -481,18 +477,11 @@ def _check_selected_atomic_numbers(self):
...
             raise AtomDataMissingError(msg)
 
     def __repr__(self):
-        return "<Atomic Data UUID={} MD5={} Lines={:d} Levels={:d}>".format(
-            self.uuid1,
-            self.md5,
-            self.lines.line_id.count(),
-            self.levels.energy.count(),
-        )
+        return f"<Atomic Data UUID={self.uuid1} MD5={self.md5} Lines={self.lines.line_id.count():d} Levels={self.levels.energy.count():d}>"
```
Comment (wkerzendorf) [inline tardis/io/atom_data/base.py:484]:
```suggestion
        return "<Atomic Data UUID={} MD5={} Lines={:d} Levels={:d}>".format(
            self.uuid1,
            self.md5,
            self.lines.line_id.count(),
            self.levels.energy.count(),
        )
```

File: tardis/io/parsers/blondin_toymodel.py [hunk 4]
```diff
@@ -159,10 +157,7 @@ def convert_blondin_toymodel(
             conversion_t_electron_rad * blondin_csv.t_electron
         )
 
-    csvy_file = "---\n{0}\n---\n{1}".format(
-        yaml.dump(blondin_dict, default_flow_style=False),
-        blondin_csv.to_csv(index=False),
-    )
+    csvy_file = f"---\n{yaml.dump(blondin_dict, default_flow_style=False)}\n---\n{blondin_csv.to_csv(index=False)}"
```
Comment (wkerzendorf) [inline tardis/io/parsers/blondin_toymodel.py:160]:
```suggestion
    csvy_file = "---\n{0}\n---\n{1}".format(
        yaml.dump(blondin_dict, default_flow_style=False),
        blondin_csv.to_csv(index=False),
    )
```

File: tardis/montecarlo/spectrum.py [hunk 5]
```diff
@@ -79,10 +74,7 @@ def flux_lambda(self):
         except AttributeError:
                 raise AttributeError(
                         'distance is required as attribute of'
-                        '{} to calculate "{}"'.format(
-                            self.__class__.__name__,
-                            'flux_lambda'
-                            )
+                        f'{self.__class__.__name__} to calculate "{"flux_lambda"}"'
```
Comment (wkerzendorf) [inline tardis/montecarlo/spectrum.py:77]:
```suggestion
                        f'{self.__class__.__name__} to calculate "flux_lambda"'
```

File: tardis/plasma/properties/atomic.py [hunk 6]
```diff
@@ -203,10 +203,7 @@ def _filter_atomic_property(self, ionization_data, selected_atoms):
             return ionization_data
         else:
             raise IncompleteAtomicData(
-                "ionization data for the ion ({}, {})".format(
-                    str(counts.index[counts.index != counts]),
-                    str(counts[counts.index != counts]),
-                )
+                f"ionization data for the ion ({str(counts.index[counts.index != counts])}, {str(counts[counts.index != counts])})"
```
Comment (wkerzendorf) [inline tardis/plasma/properties/atomic.py:206]:
```suggestion
                "ionization data for the ion ({}, {})".format(
                    str(counts.index[counts.index != counts]),
                    str(counts[counts.index != counts]),
                )counts])})"
```

File: tardis/plasma/properties/atomic.py [hunk 7]
```diff
@@ -206,7 +206,7 @@ def _filter_atomic_property(self, ionization_data, selected_atoms):
                 "ionization data for the ion ({}, {})".format(
                     str(counts.index[counts.index != counts]),
                     str(counts[counts.index != counts]),
-                )
+                )counts])})"
```
Comment (wkerzendorf) [inline tardis/plasma/properties/atomic.py:209]:
```suggestion
                )
```

File: tardis/tests/integration_tests/conftest.py [hunk 8]
```diff
@@ -48,14 +48,7 @@ def pytest_terminal_summary(terminalreporter):
...
             "-",
-            "Generated reference data: {0}".format(
-                os.path.join(
-                    terminalreporter.config.integration_tests_config[
-                        "reference"
-                    ],
-                    tardis_githash[:7],
-                )
-            ),
+            f"Generated reference data: {os.path.join(terminalreporter.config.integration_tests_config['reference'],tardis_githash[:7],)}",
```
Comment (wkerzendorf) [inline tardis/tests/integration_tests/conftest.py:51]:
```suggestion
            "Generated reference data: {0}".format(
                os.path.join(
                    terminalreporter.config.integration_tests_config[
                        "reference"
                    ],
                    tardis_githash[:7],
                )
            ),
```

**Comment by wkerzendorf** [comment]
looks - good at christian + Andreas into Co-authored

File: tardis/io/atom_data/base.py [hunk 9]
```diff
@@ -277,10 +273,10 @@ def _check_related(self):
...
             if len(check_list) != 0 and len(check_list) != len(group):
                 raise AtomDataMissingError(
-                    "The following dataframes from the related group [{0}] "
-                    "were not given: {1}".format(
-                        ", ".join(group), ", ".join(check_list)
-                    )
+                    group_str = ', '.join(group)
+                    check_list_str = ', '.join(check_list)
+                    f"The following dataframes from the related group [{group_str}] "
+                    f"were not given: {check_list_str}"
```
Comment (afloers) [inline tardis/io/atom_data/base.py:279]:
```suggestion
                    f"The following dataframes from the related group [{group_str}] "
                    f"were not given: {check_list_str}"
```

File: tardis/io/atom_data/base.py [hunk 10]
```diff
@@ -277,10 +273,10 @@ def _check_related(self):
 
             if len(check_list) != 0 and len(check_list) != len(group):
                 raise AtomDataMissingError(
```
Comment (afloers) [inline tardis/io/atom_data/base.py:275]:
```suggestion
                group_str = ', '.join(group)
                check_list_str = ', '.join(check_list)
                raise AtomDataMissingError(
```

**Pull request closed without merging**